library/core/src/test/java/com/google/android/exoplayer2/ExoPlayerTest.java

```diff
@@ -3249,6 +3249,77 @@ public void run(SimpleExoPlayer player) {
     assertThat(positionWhenReady.get()).isEqualTo(10);
   }
 
+  @Test
+  public void trackSelectionDoesNotResetToDefaultPeriodPosition()
+      throws Exception {
+    FakeMediaSource mediaSource =
+        new FakeMediaSource(new FakeTimeline(), ExoPlayerTestRunner.VIDEO_FORMAT, ExoPlayerTestRunner.AUDIO_FORMAT);
+    FakeTrackSelector trackSelector = new FakeTrackSelector();
+    FakeRenderer videoRenderer = new FakeRenderer(C.TRACK_TYPE_VIDEO);
+    FakeRenderer audioRenderer = new FakeRenderer(C.TRACK_TYPE_AUDIO);
+    MediaItem mediaItem =
+        new MediaItem.Builder().setUri(Uri.EMPTY).build();
+    final long windowDefaultStartPositionUs = 5 * C.MICROS_PER_SECOND;
+    final AtomicLong expectedPosition = new AtomicLong(C.TIME_UNSET);
+    final int lastSeekPosition = 0;
+
+    Timeline liveTimeline =
+        new SinglePeriodTimeline(
+            /* presentationStartTimeMs= */ C.TIME_UNSET,
+            /* windowStartTimeMs= */ C.TIME_UNSET,
+            /* elapsedRealtimeEpochOffsetMs= */ C.TIME_UNSET,
+            /* periodDurationUs= */ 10 * C.MICROS_PER_SECOND,
+            /* windowDurationUs= */ 10 * C.MICROS_PER_SECOND,
+            /* windowPositionInPeriodUs= */ 0,
+            /* windowDefaultStartPositionUs= */ windowDefaultStartPositionUs,
+            /* isSeekable= */ true,
+            /* isDynamic= */ true,
+            /* manifest= */ null,
+            mediaItem,
+            mediaItem.liveConfiguration);
+
+    ActionSchedule actionSchedule =
+        new ActionSchedule.Builder(TAG)
+            .pause()
+            .waitForPlaybackState(Player.STATE_BUFFERING)
+            // Finish preparation.
+            .executeRunnable(() -> mediaSource.setNewSourceInfo(liveTimeline))
+            .waitForTimelineChanged()
+            .waitForPlaybackState(Player.STATE_READY)
+            .play()
+            .playUntilPosition(0, windowDefaultStartPositionUs)
+            .pause()
+            .seekAndWait(lastSeekPosition) // paused at this position, should not change from here
+            .executeRunnable(() -> {
+              DefaultTrackSelector.Parameters parameters = trackSelector.getParameters();
+              trackSelector.setParameters(parameters.buildUpon()
+                  .setMaxAudioBitrate(0)    // Cause track selection change
+                  .build()
+              );
+            })
```

Comment:
Playback starts, reaches the default window start and is paused there.  Issue a track selection that changes parameters so as to invalidate the current selection.